src/vmspawn/vmspawn-util.c

```diff
@@ -236,3 +253,35 @@ int find_qemu_binary(char **ret_qemu_binary) {
 
         return find_executable(qemu_arch_specific, ret_qemu_binary);
 }
+
+int machine_cid(unsigned int *ret_machine_cid) {
+        int r;
+        uint64_t cid;
+        _cleanup_close_ int vfd = -EBADF;
+
+        /* Enumerate all machine CIDs in [3..UINT_MAX) until we find a free CID.
+         * if there are no free addresses return SYNTHETIC_ERRNO(EADDRNOTAVAIL)
+         * if ioctl returns an unexpected error forward that
+         */
+
+        r = open("/dev/vhost-vsock", O_RDWR);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to open /dev/vhost-vsock as read/write: %m");
+        vfd = r;
```

Comment:
```suggestion
        vfd = open("/dev/vhost-vsock", O_RDWR|O_CLOEXEC);
        if (vfd < 0)
                return log_debug_errno(errno, "Failed to open /dev/vhost-vsock as read/write: %m");
```

Reply:
(Note that all fds opened by us should have O_CLOEXEC set)

Reply:
I have a vague memory that this caused an error at some point during development, I will go and retest this.